cmd/server/cadence/server.go

```diff
@@ -222,15 +222,32 @@ func (s *server) startService() common.Daemon {
 			log.Fatalf("not able to find advanced visibility store in config: %v", advancedVisStoreKey)
 		}
 
-		params.ESConfig = advancedVisStore.ElasticSearch
+		// there are 3 circumstances:
+		// 1. advanced visibility store == elasticsearch, use ESClient and visibilityDualManager
+		// 2. advanced visibility store == pinot and in process of migration, use ESClient, PinotClient and and visibilityTripleManager
+		// 3. advanced visibility store == pinot and not migrating, use PinotClient and visibilityDualManager
 		if params.PersistenceConfig.AdvancedVisibilityStore == common.PinotVisibilityStoreName {
-			// components like ESAnalyzer is still using ElasticSearch
-			// The plan is to clean those after we switch to operate on Pinot
-			esVisibilityStore, ok := s.cfg.Persistence.DataStores[common.ESVisibilityStoreName]
-			if !ok {
-				log.Fatalf("Missing Elasticsearch config")
+			if advancedVisStore.Pinot.Migration.Enabled {
```

Comment:
`advancedVisStore.Pinot.Migration.Source` is defined but not used. we just assume it's a migration from ES to Pinot, right?

Reply:
Yeah, that is planned for future use. visibility_triple_manager will need more refactor to be more generic so we can migrate between any 2 visibility stores